### ISSUE TITLE
element-reference scanner does not detect custom elements via `is` attribute

### ISSUE THREAD

**Issue by MaKleSoft** [Polymer/polymer-analyzer#591]
```html
<form is="iron-form"></form>
```
Should be detected as `iron-form` reference, no? Currently this isn't the case.

**Comment by rictic** [comment]
Yes? We're gently discouraging using `is` because while all browsers are on board with the rest of custom elements, not everyone's on board with `is`. So I'm not 100% sure if we want to support this in the element reference scanner or not.

We'd accept a PR that added a field for `is` value and a getter that returns either the `is` tagname or the element tagname if there is no `is`.

**Comment by MaKleSoft** [comment]
> We're gently discouraging using `is`

I wasn't aware of that. Maybe you should update the documentation to reflect that? While I've seen some examples using `<dom-repeat>`, for example, most of the documentation uses `<template is="dom-repeat">`. Same thing for `iron-form`.

**Comment by rictic** [comment]
This is something we're recommending for 2.0, and we've got documentation to that effect a few places, e.g. https://www.polymer-project.org/2.0/docs/upgrade#wrap-custom-style-elements

`<iron-form>` in 2.0 will wrap a form rather than extending it. The docs need to be updated though: https://github.com/PolymerElements/iron-form/pull/172/files

dom-repeat and dom-if are so widely used that we're gonna support using them with is= in polymer element templates but I don't think we support that for other elements.

**Comment by MaKleSoft** [comment]
Thanks for clearing things up! In light of that, I can live without `is` support. I'm ok with closing this issue. Or do you want to leave it open for reference?

Btw, going forward, what is the "correct way" of using `dom-repeat` and `dom-if`? Is it
```html
<dom-repeat items="[[ items ]]">
    <template>
    </template>
</dom-repeat>
```
Or can you omit the template?

**Comment by justinfagnani** [comment]
Given that `is=` is in the spec and still viable for 1.0, we should definitely fix this.
